mlflow/pyfunc/__init__.py

```diff
@@ -1127,6 +1128,13 @@ def load_model(
 
     model_meta = Model.load(os.path.join(local_path, MLMODEL_FILE_NAME))
 
+    if model_meta.metadata and model_meta.metadata.get(MLFLOW_MODEL_IS_EXTERNAL, False) is True:
+        raise MlflowException(
+            "This model's artifacts are external and are not stored in the model directory."
+            " This model cannot be loaded with MLflow.",
+            BAD_REQUEST,
+        )
```

Comment:
mlflow.pyfunc.load_model() won't work for external logged models, so we raise a clear error message

Reply:
This `MLFLOW_MODEL_IS_EXTERNAL` metadata will also help the model registry identify that a model is external, e.g. to ensure that we don't display "serve this model" or signature info in the Databricks UC Model Registry.